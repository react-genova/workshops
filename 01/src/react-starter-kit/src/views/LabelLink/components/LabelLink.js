//@flow
import React from 'react';
import styled from 'styled-components';

type Props = {
    title:string,
    description:string,
    url:string,
    onClick:(url:string)=>void
};

const Container = styled.div`
    display: 'flex';
    flex-direction: column;
    font-family: Arial, Helvetica, sans-serif;
`;

const Title = styled.div`
    color: black;
    font-size: 1.3em;
    font-weight: bold;
    cursor: pointer;
`;

const Description = styled.div`
    color: darkgray;
`;

const LabelLink = ({title, description, url, onClick}:Props) => (
    <Container>
        <Title onClick={() => onClick(url)} title={url}>{title}</Title>
        <Description>{description}</Description>
    </Container>
);

export default LabelLink;