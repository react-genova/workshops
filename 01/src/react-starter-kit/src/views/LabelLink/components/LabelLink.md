### Simple label ###

```js
<LabelLink title="Google" description="One of the best search engine" url="http://www.google.it/" onClick={()=>null}/>
```

### On click behaviour ###

```js
initialState = { clickedUrl: 'none' };
function onClick(clickedUrl) {setState({clickedUrl})};
<div>
<div style={{display:'flex', justifyContent:'space-evenly'}}>
    <LabelLink title="Google" description="One of the best search engine" url="http://www.google.it/" onClick={onClick}/>
    <LabelLink title="Duck Duck Go" description="Another search engine" url="https://duckduckgo.com/" onClick={onClick}/>
</div>
<div style={{paddingTop:'1.4em', fontSize:'1.2em'}}>
    <span>You clicked on: </span>
    <span style={{fontWeight:'bold', color:'green'}}>{state.clickedUrl}</span>
</div>
</div>
```