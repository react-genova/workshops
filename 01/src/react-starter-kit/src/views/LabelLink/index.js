//@flow
export {default} from './components/LabelLink';