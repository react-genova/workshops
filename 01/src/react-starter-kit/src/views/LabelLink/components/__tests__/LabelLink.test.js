//@flow
import React from 'react';
import LabelLink from '../LabelLink';
import { shallow } from 'enzyme';

describe('LabelLink component', () => {

    it('simple label snapshot', () => {
        const onClick = (url:string)=>{url.toLocaleLowerCase();};
        const component = shallow(<LabelLink title="Google" description="One of the best search engine" url="http://www.google.it/" onClick={onClick}/>);
        expect(component).toMatchSnapshot();
    });

});