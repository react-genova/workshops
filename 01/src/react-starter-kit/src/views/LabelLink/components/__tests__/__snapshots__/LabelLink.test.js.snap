// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelLink component simple label snapshot 1`] = `
<styled.div>
  <styled.div
    onClick={[Function]}
    title="http://www.google.it/"
  >
    Google
  </styled.div>
  <styled.div>
    One of the best search engine
  </styled.div>
</styled.div>
`;
